Add CI/CD docs for MI (#3474)
wso2/docs-apim: en/docs/install-and-setup/setup/mi-setup/deployment/mi-cicd-overview.md

# Building a Jenkins CI/CD Pipeline for Micro Integrator

## Overview

> This is a guide of a reference implementation on setting up a CI/CD process for Micro Integrator. This guide contains two parts.

Reviewer: Can we add some information on when someone would do this or in what exact scenario? It's better to have that kind of content in the overview.

Author: Added some info. Please review

> 1. VM based CI/CD

Reviewer: Let's add links to the subtopics.

Author: Done

2. Kubernetes based CI/CD 

## Phases of SDLC

> Considering a deployment with having three typical environments as follows:

Reviewer: ```suggestion
Let's consider a deployment that has three typical environments as follows:
```

*   Developers use the Dev environment to develop their code and execute developer testing.
*   Staging (Quality Assurance) uses the Test environment for functional testing.
*   Actual consumers use the production environment

## Project structure

Main project should be an Integration project. Integration project can contain the multiple sub modules as follows. 

*   ESB Configs module
*   Data Source Configs module
*   Data Service Configs module
*   Registry Resource module
*   Mediator module
*   Composite Exporter module
*   Connector Exporter module
*   Docker Exporter module
*   Kubernetes Exporter module

> Please note that you are not allowed to create Nested Integration projects. Also for CI/CD, it is recommended to maintain one git repo per Integration project.

Reviewer: ```suggestion
Please note that you are not allowed to create Nested Integration projects. Also, for CI/CD, it is recommended to maintain one GitHub repository per Integration project.
```
